src/main/kotlin/org/rust/ide/formatter/impl/spacing.kt

```diff
@@ -268,6 +272,11 @@ private fun SpacingContext.blockMustBeMultiLine(): Boolean {
     }
 }
 
+private fun SpacingContext.needsBlankLineBetweenItems(): Boolean {
+    // Allow to keep consecutive runs of `use`, `const` or `static` items without blank lines
+    return !(elementType1 == elementType2 && elementType1 in ONE_LINE_ITEMS)
```

Comment:
Have you tried `ncPsi1.elementType` instead of `elementType1` and similarly for 2 ? I believe this should do the work. `nc` prefix comes from `non-comment`, and they are determined by `SpacingContext#omitCommentBlocks`. Though I must admit that comment handling is pretty poor now.